### PR TITLE
[v6] Refuse to use keys without key flags, add `config.allowMissingKeyFlags`

### PR DESCRIPTION
**Pull request by larabr**
Key flags are needed to restrict key usage to specific purposes: https://www.ietf.org/archive/id/draft-ietf-openpgp-crypto-refresh-10.html#section-5.2.3.29 . Some older keys (e.g. from OpenPGP.js v0) do not declare any key flags. In previous OpenPGP.js versions, we've allowed such keys to be used for any operation for which they were compatible. This behaviour has now changed, and these keys are not allowed to be used for any operation.

The setting  `config.allowMissingKeyFlags` has been added to selectively revert to the past behaviour.

### CODE REVIEW

File: src/config/config.js [hunk 1]
```diff
@@ -166,7 +166,14 @@ export default {
    * @property {Boolean} allowInsecureDecryptionWithSigningKeys
    */
   allowInsecureVerificationWithReformattedKeys: false,
-
+  /**
+   * Allow using keys that do not have any key flags set.
+   * Key flags are needed to restrict key usage to specific purposes: for instance, a signing key could only be allowed to certify other keys, and not sign messages
+   * (see https://www.ietf.org/archive/id/draft-ietf-openpgp-crypto-refresh-10.html#section-5.2.3.29).
+   * Some older keys (e.g. from OpenPGP.js v1) do not declare any key flags, which means they are not allowed to be used for any operation.
```
Comment (larabr) [inline src/config/config.js:173]:
```suggestion
   * Some older keys do not declare any key flags, which means they are not allowed to be used for any operation.
```

File: src/key/helper.js [hunk 2]
```diff
@@ -355,28 +355,49 @@ export function sanitizeKeyOptions(options, subkeyDefaults = {}) {
...
+    case enums.publicKey.ecdsa:
+    case enums.publicKey.ed25519Legacy:
+    case enums.publicKey.ed25519: {
+      if (!signature.keyFlags && !config.allowMissingKeyFlags) {
+        throw new Error('None of the key flags is set: consider passing `config.allowMissingKeyFlags`');
+      }
+
+      return !signature.keyFlags ||
+        (signature.keyFlags[0] & enums.keyFlags.signData) !== 0;
+    }
```
Comment (twiss) [inline src/key/helper.js:372]:
```suggestion
    }
    default:
      return false;
```

File: src/key/helper.js [hunk 3]
```diff
@@ -355,28 +355,49 @@ export function sanitizeKeyOptions(options, subkeyDefaults = {}) {
...
+    case enums.publicKey.ecdh:
+    case enums.publicKey.x25519: {
+      if (!signature.keyFlags && !config.allowMissingKeyFlags) {
+        throw new Error('None of the key flags is set: consider passing `config.allowMissingKeyFlags`');
+      }
+
+      return !signature.keyFlags ||
+        (signature.keyFlags[0] & enums.keyFlags.encryptCommunication) !== 0 ||
+        (signature.keyFlags[0] & enums.keyFlags.encryptStorage) !== 0;
+    }
```
Comment (twiss) [inline src/key/helper.js:390]:
```suggestion
    }
    default:
      return false;
```

File: src/key/helper.js [hunk 4]
```diff
@@ -355,35 +355,70 @@ export function sanitizeKeyOptions(options, subkeyDefaults = {}) {
...
+    case enums.publicKey.ecdsa:
+    case enums.publicKey.ed25519Legacy:
+    case enums.publicKey.ed25519: {
+      if (!signature.keyFlags && !config.allowMissingKeyFlags) {
+        throw new Error('None of the key flags is set: consider passing `config.allowMissingKeyFlags`');
+      }
+
+      return !signature.keyFlags ||
+        (signature.keyFlags[0] & enums.keyFlags.signData) !== 0;
+    }
```
Comment (twiss) [inline src/key/helper.js:372]: Sorry for missing these last time (and being very nitpicky ^.^), but these brackets are also not necessary. They're only necessary if you need a block scope, e.g. if you're creating block-scoped variables.
```suggestion
    case enums.publicKey.ed25519:
      if (!signature.keyFlags && !config.allowMissingKeyFlags) {
        throw new Error('None of the key flags is set: consider passing `config.allowMissingKeyFlags`');
      }

      return !signature.keyFlags ||
        (signature.keyFlags[0] & enums.keyFlags.signData) !== 0;
```

File: src/key/helper.js [hunk 5]
```diff
@@ -355,35 +355,70 @@ export function sanitizeKeyOptions(options, subkeyDefaults = {}) {
...
+    case enums.publicKey.ecdh:
+    case enums.publicKey.x25519: {
+      if (!signature.keyFlags && !config.allowMissingKeyFlags) {
+        throw new Error('None of the key flags is set: consider passing `config.allowMissingKeyFlags`');
+      }
+
+      return !signature.keyFlags ||
+        (signature.keyFlags[0] & enums.keyFlags.encryptCommunication) !== 0 ||
+        (signature.keyFlags[0] & enums.keyFlags.encryptStorage) !== 0;
+    }
```
Comment (twiss) [inline src/key/helper.js:392]:
```suggestion
    case enums.publicKey.x25519:
      if (!signature.keyFlags && !config.allowMissingKeyFlags) {
        throw new Error('None of the key flags is set: consider passing `config.allowMissingKeyFlags`');
      }

      return !signature.keyFlags ||
        (signature.keyFlags[0] & enums.keyFlags.encryptCommunication) !== 0 ||
        (signature.keyFlags[0] & enums.keyFlags.encryptStorage) !== 0;
```

**Review by twiss — APPROVED** [review]

File: src/key/helper.js (hunk 4 above)
Comment (larabr) [reply src/key/helper.js:372]: I know they are not needed, but it's not a big deal to add them...? I just find it confusing to have a lot of code without brackets.

Comment (twiss) [reply src/key/helper.js:372]: Er, OK, but I think it's more important to have a consistent code style. JS cases normally don't have brackets, and most other code in the library doesn't have them, so we should only add them when needed.

Comment (larabr) [reply src/key/helper.js:372]: Ok, but then I'll add a linting rule for this, so that it's automatically enforced

**Pull request merged**